### PR TITLE
Image Visualizations

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

PR implements visualizations for the image library: https://github.com/enso-org/enso/pull/1450

Changelog:
- feat: visualization for base64 encoded images
- feat: histogram visualization supports new `data.bins` field with precomputed bins

Example:
2021-03-18-150933_1461x653_scrot

### Checklist
Please include the following checklist in your PR:

- [x] The `CHANGELOG.md` was updated with the changes introduced in this PR.
- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has automatic tests where possible.
- [x] All code has been profiled where possible.
- [x] All code has been manually tested in the IDE.
- [x] All code has been manually tested in the "debug/interface" scene.
- [x] All code has been manually tested by the PR owner against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).
- [ ] All code has been manually tested by at least one reviewer against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).

### CODE REVIEW

File: src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/imageBase64.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            parsedData = JSON.parse(data)
+        }
+        const mediaType = parsedData.mediaType || 'image/png'
+        const base64 = parsedData.base64
+
+        const width = this.dom.getAttributeNS(null, 'width')
+        const height = this.dom.getAttributeNS(null, 'height')
+        const tabElem = document.createElement('div')
+        tabElem.setAttributeNS(null, 'id', 'vis-image-view')
+        tabElem.setAttributeNS(null, 'class', 'scrollable')
```
Comment (BinarySoftware) [inline src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/imageBase64.js]: I don't think that it should use `scrollable` class, rather `visualization`. Scrollable only adds scrollbars if it overflows, but visualization adds rounded corners and i think drop shadow too?

Comment (4e6) [reply src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/imageBase64.js]: Sure, I added it by analogy with other visualizations. Removed :+1:

File: src/rust/ide/view/graph-editor/src/builtin/visualization/java_script.rs [hunk 2]
```diff
@@ -4,8 +4,6 @@
 
 use crate::component::visualization;
 
-
-
```
Comment (farmaazon) [inline src/rust/ide/view/graph-editor/src/builtin/visualization/java_script.rs]: We keep 3 empty lines before section headers. Please restore those two

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,12 @@
   will be lost. In this build we added notification in statusbar to signalize
   that the connection was lost and IDE must be restarted. In future IDE will try
   to automatically reconnect.
+- [Image visualizations][1367]. Visualizations for the Enso Image library. New
+  image visualization can display a JSON with an image encoded as a base64
+  string. Histogram visualization has a new feature allowing to display the
```
Comment (farmaazon) [inline CHANGELOG.md]: The "JSON with an image encoded as a base64" is not understandable for the user: they do not know in what format visualizations receive data. They just see visualization attached to nodes returning some enso types. The exact format is defined by visualization themselfs, by setting proper preprocessor code. The useful information is: you can display Image type and string with image encoded in base64.

File: src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/imageBase64.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+class ImageBase64Visualization extends Visualization {
+    static inputType = 'Any'
```
Comment (farmaazon) [inline src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/imageBase64.js]: FYI once I merge my current work on #837 the inputType should be full path for `Image` and `String`.

**Review by farmaazon — CHANGES_REQUESTED** [review]
There are two main issues:
1. All our visualizations has rounded corners, but the Image Vis has sharp corners.
2. The visualization should allow zooming and panning.
3. I'd consider setting own preprocessor, instead of relying on the default one. For example, the preprocessor could explicitly check if the value's type is `Image` or `String`.

**Review by farmaazon — APPROVED** [review]
@4e6 @BinarySoftware Ok, after a bunch of calls we decided, that this PR can be merged without panning and zooming, and also setting own preprocessor is not necessary. But please fix the comment for CHANGELOG before merge + change inputType to `Image | String` (but with full qualified names)

**Comment by BinarySoftware** [comment]
> @4e6 @BinarySoftware Ok, after a bunch of calls we decided, that this PR can be merged without panning and zooming, and also setting own preprocessor is not necessary. But please fix the comment for CHANGELOG before merge + change inputType to `Image | String` (but with full qualified names)

Input type has already been changed. Will change changelog rn

**Comment by 4e6** [comment]
Thanks for handling this for me. I was preparing stuff for the upcoming release, and had to abandon this PR for a while.

**Review by wdanilo — APPROVED** [review]

**Pull request merged**